### PR TITLE
[action] enable tx container

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/web3server.go [hunk 1]
```diff
@@ -502,27 +500,30 @@ func (svr *web3Handler) sendRawTransaction(in *gjson.Result) (interface{}, error
 	if err != nil {
 		return nil, err
 	}
-	if g := cs.Genesis(); g.IsToBeEnabled(cs.TipHeight()) {
-		if strings.HasPrefix(rawString, "0x") || strings.HasPrefix(rawString, "0X") {
-			rawString = rawString[2:]
+	if g := cs.Genesis(); g.IsVanuatu(cs.TipHeight()) {
+		elp, err := svr.stakingRewardingTxToEnvelope(tx)
```
Comment (envestcc) [inline api/web3server.go:504]: can reuse `ethTxToEnvelope` instead

File: action/tx_container.go [hunk 2]
```diff
@@ -61,20 +163,24 @@ func (etx *txContainer) proto() *iotextypes.TxContainer {
 	}
 }
 
-func (etx *txContainer) loadProto(pbAct *iotextypes.TxContainer) error {
+func (etx *txContainer) LoadProto(pbAct *iotextypes.ActionCore) error {
```
Comment (envestcc) [inline action/tx_container.go:166]: seems not used

File: action/blob_data.go [hunk 3]
```diff
@@ -166,6 +166,11 @@ func (tx *BlobTxData) ValidateSidecar() error {
 	if permitted := params.MaxBlobGasPerBlock / params.BlobTxBlobGasPerBlob; size > permitted {
 		return errors.Errorf("too many blobs in transaction: have %d, permitted %d", size, params.MaxBlobGasPerBlock/params.BlobTxBlobGasPerBlob)
 	}
```
Comment (envestcc) [inline action/blob_data.go:168]: can move to SanityCheck

File: action/tx_container.go [hunk 4]
```diff
@@ -11,25 +11,131 @@ import (
...
+	// TODO
+	return "", true
+}
+
+func (etx *txContainer) Size() uint32 {
+	// TODO
+	return 0
+}
+
+func (etx *txContainer) Action() Action { return etx }
```
Comment (envestcc) [inline action/tx_container.go:107]: can we return the unfolded tx in Action()? thus will keep consistent in workingset

File: action/tx_container.go (hunk 2 above)
Comment (dustinxie) [reply action/tx_container.go:166]: used in `sealedenvelope.go`

File: action/tx_container.go [hunk 5]
```diff
@@ -11,25 +11,131 @@ import (
...
+
+func (etx *txContainer) ToEthTx(evmNetworkID uint32, encoding iotextypes.Encoding) (*types.Transaction, error) {
+	return etx.tx, nil
+}
+
+func (etx *txContainer) ProtoForHash() *iotextypes.ActionCore {
+	// TODO
```
Comment (envestcc) [inline action/tx_container.go:114]: same

File: action/tx_container.go [hunk 6]
```diff
@@ -11,25 +11,131 @@ import (
...
+	return nil
+}
+
+func (etx *txContainer) Proto() *iotextypes.ActionCore {
+	// TODO
```
Comment (envestcc) [inline action/tx_container.go:119]: same

File: action/blob_data.go (hunk 3 above)
Comment (CoderZhi) [reply action/blob_data.go:168]: They could be moved into `verifySidecar`

**Review by CoderZhi — APPROVED** [review]

File: action/tx_container.go [hunk 7]
```diff
@@ -130,6 +215,12 @@ func (etx *txContainer) IntrinsicGas() (uint64, error) {
 	return gas, nil
 }
 
+func (etx *txContainer) SetNonce(n uint64) {}
```
Comment (envestcc) [inline action/tx_container.go:218]: also shouldn't call it?

**Review by envestcc — APPROVED** [review]

**Pull request merged**